RF24Network.cpp

```diff
@@ -700,10 +700,23 @@ template<class radio_t>
 bool ESBNetwork<radio_t>::write(RF24NetworkHeader& header, const void* message, uint16_t len, uint16_t writeDirect)
 {
 
+#if defined(NRF52_RADIO_LIBRARY)
+    max_frame_size = (uint8_t)NRF_RADIO->PCNF1;
+    /*if(radio.enableEncryption == true){
+      max_frame_size -= CCM_IV_SIZE + CCM_COUNTER_SIZE + CCM_MIC_SIZE;
+    }*/
```

Comment:
Still not sure how to approach this without adding a variable into RF24. I tried using the `radio_t == nrf_to_nrf` but it throws an error.

Getting an error `705:17: error: expected unqualified-id before '==' token`
  `705 |     if (radio_t == nrf_to_nrf) {`

Reply:
I figured that might not work. We might need to define the function for each specialization of the template...
```cpp
EsbNetwork<RF24>::write(/* ... */) {
 // anything specific to RF24 here

 // call template (generic) function here
 writeGeneric(/* ... */);
}

EsbNetwork<nrf_to_nrf>::write(/* ... */) {
 // anything specific to nRF5x here

 // call template (generic) function here
 writeGeneric(/* ... */);
}

template<class radio_t>
EsbNetwork<radio_t>::writeGeneric(/* ... */) {
  // original code goes here
}
```

This too might be erroneous as the compiler isn't told what specializations are supported (in RF24Network.h) until it compiles the RF24Network.cpp https://github.com/nRF24/RF24Network/blob/731a36864daa72a605eb170aed8b29493ecfd6f4/RF24Network.cpp#L1316-L1320

Reply:
See also [this SO answer](https://stackoverflow.com/questions/43045457/c-templated-class-with-different-behaviour-depending-on-type).

Reply:
So I went with function overloading, I think I got it right.

Reply:
I'll reboot into Linux tomorrow (or later today) and toy around with this branch. I'm curious if this can be done without the `RadioTag` struct. It might be ok due to implicit zero-cost abstraction (syntax pattern which informs the compiler that runtime overhead about variant data types is not needed -- a very common practice in rust).

Reply:
removing the radioTag seemed simple, but I haven't tested it yet.